webapp/pages/settings/my-social-media.vue

```diff
@@ -1,61 +1,101 @@
+                <img :src="link.favicon" alt="Link:" height="16" width="16" />
+                {{ link.url }}
+              </a>
+              <span class="divider">|</span>
+              <a name="edit" @click="handleEditSocialMedia(link)">
+                <ds-icon
+                  :aria-label="$t('actions.edit')"
+                  class="icon-button"
+                  name="edit"
+                  :title="$t('actions.edit')"
+                />
+              </a>
+              <a name="delete" @click="handleDeleteSocialMedia(link)">
+                <ds-icon
+                  :aria-label="$t('actions.delete')"
+                  class="icon-button"
+                  name="trash"
+                  :title="$t('actions.delete')"
+                />
+              </a>
+            </template>
+          </ds-list-item>
+        </ds-list>
+      </ds-space>
+
+      <ds-space margin-top="base">
         <ds-input
-          v-model="value"
+          v-if="!editingLink.id"
+          id="addSocialMedia"
+          model="socialMediaUrl"
+          type="text"
           :placeholder="$t('settings.social-media.placeholder')"
-          name="social-media"
-          :schema="{ type: 'url' }"
         />
-      </div>
-      <ds-space margin-top="base">
-        <div>
-          <ds-button primary @click="handleAddSocialMedia">
-            {{ $t('settings.social-media.submit') }}
+        <ds-space margin-top="base">
+          <ds-button primary :disabled="disabled">
+            {{ editingLink.id ? $t('actions.save') : $t('settings.social-media.submit') }}
+          </ds-button>
+          <ds-button v-if="editingLink.id" id="cancel" ghost @click="handleCancel()">
+            {{ $t('actions.cancel') }}
           </ds-button>
-        </div>
+        </ds-space>
       </ds-space>
-    </ds-space>
-  </ds-card>
+    </ds-card>
+  </ds-form>
 </template>
+
 <script>
+import unionBy from 'lodash/unionBy'
 import gql from 'graphql-tag'
 import { mapGetters, mapMutations } from 'vuex'
 
 export default {
   data() {
     return {
-      value: '',
+      formData: {
+        socialMediaUrl: '',
+      },
+      formSchema: {
+        socialMediaUrl: {
+          type: 'url',
+          message: this.$t('common.validations.url'),
+        },
+      },
+      disabled: true,
+      editingLink: {},
     }
   },
   computed: {
     ...mapGetters({
       currentUser: 'auth/user',
     }),
     socialMediaLinks() {
+      const domainRegex = /^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g
       const { socialMedia = [] } = this.currentUser
-      return socialMedia.map(socialMedia => {
-        const { id, url } = socialMedia
-        const matches = url.match(/^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
-        const [domain] = matches || []
+      return socialMedia.map(({ id, url }) => {
+        const [domain] = url.match(domainRegex) || []
         const favicon = domain ? `${domain}/favicon.ico` : null
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-17T11:31:33Z" title="Wednesday, July 17th 2019, 1:31:33 pm +02:00">Jul 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -1,61 +1,101 @@
 <template>
-  <ds-card :header="$t('settings.social-media.name')">
-    <ds-space v-if="socialMediaLinks" margin-top="base" margin="x-small">
-      <ds-list>
-        <ds-list-item v-for="link in socialMediaLinks" :key="link.id">
-          <a :href="link.url" target="_blank">
-            <img :src="link.favicon | proxyApiUrl" alt="Social Media link" width="16" height="16" />
-            {{ link.url }}
-          </a>
-          &nbsp;&nbsp;
-          <span class="layout-leave-active">|</span>
-          &nbsp;&nbsp;
-          <ds-icon name="edit" class="layout-leave-active" />
-          <a name="delete" @click="handleDeleteSocialMedia(link)">
-            <ds-icon name="trash" />
-          </a>
-        </ds-list-item>
-      </ds-list>
-    </ds-space>
-    <ds-space margin-top="base">
-      <div>
+  <ds-form
+    v-model="formData"
+    :schema="formSchema"
+    @input="handleInput"
+    @input-valid="handleInputValid"
+    @submit="handleSubmitSocialMedia"
+  >
+    <ds-card :header="$t('settings.social-media.name')">
+      <ds-space v-if="socialMediaLinks" margin-top="base" margin="x-small">
+        <ds-list>
+          <ds-list-item v-for="link in socialMediaLinks" :key="link.id">
+            <ds-input
+              v-if="editingLink.id === link.id"
+              id="editSocialMedia"
+              model="socialMediaUrl"
+              type="text"
+              :placeholder="$t('settings.social-media.placeholder')"
+            />
+
+            <template v-else>
+              <a :href="link.url" target="_blank">
+                <img :src="link.favicon" alt="Link:" height="16" width="16" />
+                {{ link.url }}
+              </a>
+              <span class="divider">|</span>
+              <a name="edit" @click="handleEditSocialMedia(link)">
+                <ds-icon
+                  :aria-label="$t('actions.edit')"
+                  class="icon-button"
+                  name="edit"
+                  :title="$t('actions.edit')"
+                />
+              </a>
+              <a name="delete" @click="handleDeleteSocialMedia(link)">
+                <ds-icon
+                  :aria-label="$t('actions.delete')"
+                  class="icon-button"
+                  name="trash"
+                  :title="$t('actions.delete')"
+                />
+              </a>
+            </template>
+          </ds-list-item>
+        </ds-list>
+      </ds-space>
+
+      <ds-space margin-top="base">
         <ds-input
-          v-model="value"
+          v-if="!editingLink.id"
+          id="addSocialMedia"
+          model="socialMediaUrl"
+          type="text"
           :placeholder="$t('settings.social-media.placeholder')"
-          name="social-media"
-          :schema="{ type: 'url' }"
         />
-      </div>
-      <ds-space margin-top="base">
-        <div>
-          <ds-button primary @click="handleAddSocialMedia">
-            {{ $t('settings.social-media.submit') }}
+        <ds-space margin-top="base">
+          <ds-button primary :disabled="disabled">
+            {{ editingLink.id ? $t('actions.save') : $t('settings.social-media.submit') }}
+          </ds-button>
+          <ds-button v-if="editingLink.id" id="cancel" ghost @click="handleCancel()">
+            {{ $t('actions.cancel') }}
           </ds-button>
-        </div>
+        </ds-space>
       </ds-space>
-    </ds-space>
-  </ds-card>
+    </ds-card>
+  </ds-form>
 </template>
+
 <script>
+import unionBy from 'lodash/unionBy'
 import gql from 'graphql-tag'
 import { mapGetters, mapMutations } from 'vuex'
 
 export default {
   data() {
     return {
-      value: '',
+      formData: {
+        socialMediaUrl: '',
+      },
+      formSchema: {
+        socialMediaUrl: {
+          type: 'url',
+          message: this.$t('common.validations.url'),
+        },
+      },
+      disabled: true,
+      editingLink: {},
     }
   },
   computed: {
     ...mapGetters({
       currentUser: 'auth/user',
     }),
     socialMediaLinks() {
+      const domainRegex = /^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g
       const { socialMedia = [] } = this.currentUser
-      return socialMedia.map(socialMedia => {
-        const { id, url } = socialMedia
-        const matches = url.match(/^(?:https?:\/\/)?(?:[^@\n])?(?:www\.)?([^:/\n?]+)/g)
-        const [domain] = matches || []
+      return socialMedia.map(({ id, url }) => {
+        const [domain] = url.match(domainRegex) || []
         const favicon = domain ? `${domain}/favicon.ico` : null
```


I really like this refactor! thank you :heart_eyes_cat:

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-07-18T06:56:40Z" title="Thursday, July 18th 2019, 8:56:40 am +02:00">Jul 18, 2019</time>_

---


😻